src/net_processing.cpp

```diff
@@ -3208,7 +3208,14 @@ bool ProcessMessage(CNode* pfrom, const std::string& msg_type, CDataStream& vRec
         pfrom->fSentAddr = true;
 
         pfrom->vAddrToSend.clear();
-        std::vector<CAddress> vAddr = connman->GetAddresses();
+        CConnman::AddrCacheNetwork use_cache;
+        Network peer_network = pfrom->addr.GetNetwork();
+        if (peer_network == NET_IPV4 || peer_network == NET_IPV6) {
+            use_cache = CConnman::ADDR_CACHE_REGULAR;
+        } else {
+            use_cache = CConnman::ADDR_CACHE_OTHER;
```

Comment:
My understand is you want to use `ADDR_CACHE_OTHER` for privacy-preserving networks, so that the caches returned from these networks  are not mapped 1-1 with caches returned from identifying networks like ipv4, is that right?  

If this is the case, I believe NET_TEREDO peers and any peers with NET_UNKNOWN will also hit the `else` case and use the same cache as Tor's?

Reply:
>My understand is you want to use ADDR_CACHE_OTHER for privacy-preserving networks, so that the caches returned from these networks are not mapped 1-1 with caches returned from identifying networks like ipv4, is that right?

Yes. I mean, ideally not just private/identifying networks. Ideally, if a node is hosted on Tor AND i2p AND ipv4, neither of them should be mapped together. But allocating a new cache for every network is probably too big of an overkill for such a niche use-case.

That's why I leave 2 caches, keeping in mind that ipv4 + ipv6 is one big group, and everything else can be grouped together. So, I guess that answers your question?

If other reviewers point out some other usecase, I'll be happy to discuss more.